### ISSUE TITLE
train error

### ISSUE THREAD

**Issue by Dyf-hue** [sunzc-sunny/RNightSeg#1]
您好，使用您的代码重训练时候出现IndexError: Target 23 is out of bounds.报错。推测可能是数据集的问题，我将nightcity+的val标注遍历了一遍。发现它的像素值并不是0-18.具体如下图：
<img width="1182" alt="微信图片_20241003164934" src="https://github.com/user-attachments/assets/07978530-3218-42cf-b665-96c4bd633cf2">
为什么官方下载的数据集标签范围不在0-18之间？希望您能解答一下，谢谢！

**Comment by sunzc-sunny** [comment]
mmsegmentation训练时需要把label映射到0-18，您可以先把标签重新映射保存后再训练。这个是我当时的映射代码，供您参考
import os
import cv2
import numpy as np
import torch
from PIL import Image
root = '../../../data/sunzc/data/NightCity/NightCity-label/label/train'

dir = '../../../data/sunzc/data/NightCity/NightCity-label/label/train_new'

os.mkdir(dir)

imgs_path_list = os.listdir(root)
len_ = len(imgs_path_list)
i = 0

for item in imgs_path_list:
    img = cv2.imread(os.path.join(root,item))
    img = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
    print(img.shape)
    img [img ==0] =255
    img [img ==1] =255
    img [img ==2] =255
    img [img ==3] =255
    img [img ==4] =255
    img [img ==5] =255
    img [img ==6] =255
    img [img ==7] =0
    img [img ==8] =1
    img [img ==9] =255
    img [img ==10] =255
    img [img ==11] =2
    img [img ==12] =3
    img [img ==13] =4
    img [img ==14] =255
    img [img ==15] =255
    img [img ==16] =255
    img [img ==17] =5
    img [img ==18] =255
    img [img ==19] =6
    img [img ==20] =7
    img [img ==21] =8
    img [img ==22] =9
    img [img ==23] =10
    img [img ==24] =11
    img [img ==25] =12
    img [img ==26] =13
    img [img ==27] =14
    img [img ==28] =15
    img [img ==29] =255
    img [img ==30] =255
    img [img ==31] =16
    img [img ==32] =17
    img [img ==33] =18

    temp = os.path.join(dir, item)

    cv2.imwrite(temp, img)
    i += 1

**Comment by Dyf-hue** [comment]
非常感谢您的回复，我尝试使用了您的代码对nightcity训练和校验标签进行了重映射，没有问题，但是再对nightcity+的校验标签重映射时候提示有问题，应该是一些像素值没有在映射字典中定义导致的，我需要将这些未定义的像素值设置成255吗？ 这些像素值的映射是有什么要求吗？刚转语义分割没多久，不太熟悉，麻烦您了。

**Comment by sunzc-sunny** [comment]
img [img > 33] = 255
    img [img < 0] = 255
    是的，应该在加这两行就可以了。
    这个映射是常用的方案，很多语义分割的数据集都需要映射，您可以参考一下mmsegmentation中的dataset_prepare.md 
[dataset_prepare.md](https://github.com/open-mmlab/mmsegmentation/blob/main/docs/en/user_guides/2_dataset_prepare.md#prepare-datasets)

**Comment by Dyf-hue** [comment]
![image](https://github.com/user-attachments/assets/cf1139d6-332e-4d6d-bd46-57ad5fc26dc6)
您使用的nightcity+是将val标注resize到512x1024吗？

**Comment by sunzc-sunny** [comment]
是的，这个操作都在dataset的config里

**Comment by Dyf-hue** [comment]
非常抱歉再次打扰您，我想知道您论文中输出的增强图片是怎么输出的？ 
使用默认的推理引擎好像只能输出掩膜？输出不了图像
![image](https://github.com/user-attachments/assets/fc897bbd-f4d7-461f-9a4d-09579d58d43f)

**Comment by sunzc-sunny** [comment]
感谢您对我们工作的关注，正常分割的任务中，为了提高推理速度，我们没有选择输出增强后的图片。但是，如果需要保存增强后的图片，需要您在推理时，运行[configs/RNightseg_deeplab_test.py](https://github.com/sunzc-sunny/RNightSeg/blob/master/configs/RNightseg_deeplab_test.py)这个config文件，具体的保存代码在[parallel_decode_test](https://github.com/sunzc-sunny/RNightSeg/blob/master/mmseg/models/decode_heads/parallel_decode_test.py)中，您可以修改其中的内容来修改保存路径和保存的内容。
